operator/internal/manifests/distributor.go

```diff
@@ -59,7 +59,8 @@ func NewDistributorDeployment(opts Options) *appsv1.Deployment {
 	l := ComponentLabels(LabelDistributorComponent, opts.Name)
 	a := commonAnnotations(opts.ConfigSHA1, opts.CertRotationRequiredAt)
 	podSpec := corev1.PodSpec{
-		Affinity: configureAffinity(l, opts.Gates.DefaultNodeAffinity),
+		Affinity:                  configureAffinity(l, opts.Gates.DefaultNodeAffinity),
+		TopologySpreadConstraints: defaultTopologySpreadConstraints(l),
```

Comment:
This definition is overwritten by and thus no node-spreading is happening when zone-aware data replication is on. See:
https://github.com/grafana/loki/blob/520e434446e4c646ecc2678f3c3b57fff69643cc/operator/internal/manifests/var.go#L146-L166